Framework/Core/include/Framework/TableTreeHelpers.h

```diff
@@ -37,6 +38,7 @@ class branchIterator
 {
 
  private:
+  arrow::MemoryPool* mpool = arrow::default_memory_pool();
```

Comment:
Please follow the Coding Guidelines of O2. In particular:

* Classes start with a capital letter e.g. `Something` unless the C++ standard recommends / does otherwise (i.e. it's fine to have an iterator being called "iterator");
* members start with m and then a capital letter when they are part of a class. In case it's more readable to have lowercase members, you probably should use a separate "data only" struct with members which are public.

The whole class should actually be cleaned up.

Reply:
On a separate note, why do you need to cache the allocator in the iterator?

Reply:
"cache the allocator in the iterator" - can you specify which part of the code you are addressing?

Reply:
The one highlighted: `arrow::MemoryPool* mpool = arrow::default_memory_pool()`